spinn_front_end_common/interface/abstract_spinnaker_base.py

```diff
@@ -3177,12 +3217,12 @@ def _recover_from_error(self, exception, exc_info, executable_targets):
                         self._placements.get_placement_on_processor(x, y, p))
                 extractor = PlacementsProvenanceGatherer()
                 extractor(self._txrx, finished_placements)
-            except Exception:
-                logger.exception("Could not read provenance")
+            except Exception as pro_e:
+                logger.exception(f"Could not read provenance due to {pro_e}")
 
         # Read IOBUF where possible (that should be everywhere)
         iobuf = IOBufExtractor(
-            self._txrx, executable_targets, self._executable_finder)
+            self._txrx, self._executable_targets, self._executable_finder)
         try:
             errors, warnings = iobuf.extract_iobuf()
```

Comment:
I think the "print" call should also go in here, otherwise it will fail because things are not defined.

Reply:
Ah no, I see now that it does define them in the except.  Odd, but OK...